libarchive/_libarchive.i

```diff
@@ -171,39 +169,107 @@ extern int archive_read_next_header2(struct archive *,
 extern const struct stat	*archive_entry_stat(struct archive_entry *);
 extern __LA_INT64_T		 archive_read_header_position(struct archive *);
 
+/*
+ * Set read options.
+ */
+/* Apply option to the format only. */
+extern int archive_read_set_format_option(struct archive *_a,
+			    const char *m, const char *o,
+			    const char *v);
+/* Apply option to the filter only. */
+extern int archive_read_set_filter_option(struct archive *_a,
+			    const char *m, const char *o,
+			    const char *v);
+/* Apply option to both the format and the filter. */
+extern int archive_read_set_option(struct archive *_a,
+			    const char *m, const char *o,
+			    const char *v);
+/* Apply option string to both the format and the filter. */
+extern int archive_read_set_options(struct archive *_a,
+			    const char *opts);
+
+/*
+ * Add a decryption passphrase.
+ */
+extern int archive_read_add_passphrase(struct archive *, const char *);
 /* data */
 extern int archive_read_data_skip(struct archive *);
 extern int archive_read_data_into_fd(struct archive *, int fd);
 
-/* FILTERS */
+#if ARCHIVE_VERSION_NUMBER < 4000000
+extern int archive_read_support_compression_all(struct archive *);
```

Comment:
When I build, I get warnings that all of these are deprecated
```
cc -c -g -fPIC  $(python3.8-config --cflags)  _libarchive_wrap.c
_libarchive_wrap.c: In function ‘_wrap_archive_read_support_compression_all’:
_libarchive_wrap.c:3864:3: warning: ‘archive_read_support_compression_all’ is deprecated [-Wdeprecated-declarations]
 3864 |   result = (int)archive_read_support_compression_all(arg1);
      |   ^~~~~~
In file included from _libarchive_wrap.c:2712:
/usr/include/archive.h:392:15: note: declared here
  392 | __LA_DECL int archive_read_support_compression_all(struct archive *)
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```
Are you seeing the same thing? Is there an updated interface that should be used? It looks like there is a check for `ARCHIVE_VERSION_NUMBER` that should be accounting for that

Reply:
Yes i'm getting them too.
Given the fact that the master branch version archive.h in libarchive repo is only   3.6.2
I think we can keep it as it is...